### DISCUSSION TITLE
Serve static files for remote access

### DISCUSSION THREAD

**Discussion by igorjacauna** [tauri-apps/tauri#4998] [Q&A] [answered]
Is it possible to get the assets directory to upload files to it and serve like static files using `tauri-plugin-localhost`? I searched on docs something to retrieve the path to assets but did not find anything, only assets resolver but its not resolve the path if I want to upload custom files in runtime and delivery them through `tauri-plugin-localhost`

I asked [this](https://github.com/tauri-apps/tauri-plugin-localhost/issues/23) on plugin repo but no answers yet

**Comment by FabianLars** [comment, marked as answer]
> Is it possible to get the assets directory to upload files to it

There is no asset directory, all the files are stored as binary data inside the executable. So for serving custom files you need to spin up your own localhost server. (You could combine this with serving the bundled assets by using the asset resolver you mentioned inside that new server)

**Comment by whyboris** [comment]
I was able to achieve this* 🤩 

Three commits that got me there:
1) https://github.com/whyboris/my-off-remote/pull/1/changes/0a09e8c76b3c6f4aeca389fcc38c82c8b891d959 -- add `axum` router
2) https://github.com/whyboris/my-off-remote/pull/1/changes/68dd998acd7be03eb6c8a1c25ee1a434c3f8cf55 -- add `assets` folder with a static `html` file and serve it with the router
3) https://github.com/whyboris/my-off-remote/pull/1/changes/31d99d77807af753a90acadc0df65883bc553215 -- change `127.0.0.1:3000` to `0.0.0.0:3000`

While serving `127.0.0.1:3000` the machine where I am coding serves the files (accessible through `localhost:3000` in the browser), but only after changing to `0.0.0.0:3000` does the app allow other devices on the same _WiFi_ connect to the app.

\* - _note_ - the setup I have is for _Tauri_ to serve static files that I include when building the application (they live inside the executable), not for serving static files dynamically from somewhere else 🤝
